helpers/common.go

```diff
@@ -27,16 +30,25 @@ func NewHelper() Impl {
 }
 
 // GetDataFromRequest will try to retrieve the Body from a given request and convert it into a string.
-func (h Impl) GetDataFromRequest(r *http.Request) (string, error) {
+func (h Impl) GetDataFromRequest(r *http.Request, contextId string) (string, error) {
 
 	// Retrieve the request body.
 	data, err := callReadAll(r.Body)
 	if err != nil {
-		log.Error(fmt.Errorf("error while retrieving the Body from a given request and converting it into a string : %s", err), nil)
+		log.ErrorC(contextId, fmt.Errorf("error while retrieving the Body from a given request and converting it into a string : %s", err), nil)
 		return "", err
 	}
 
 	// Convert the request body into a string and pass it to the Kafka Service for publishing.
 	strData := string(data)
 	return strData, nil
 }
+
+//GetRequestIdFromHeader gets X-Request-Id from header and use it as a context id for logging
+func (h Impl) GetRequestIdFromHeader(r *http.Request) (string, error) {
+	requestID := r.Header.Get(xRequestId)
+	if requestID == "" {
+		return "Context ID not found", fmt.Errorf("unable to extract request ID")
```

Comment:
Instead of returning a full msg here, just return a dummy ID such as "contextID" along with the error message. That way we can still use the contextID in logging but it's just a placeholder, you can then review the logs when you see the placeholder to find out why it failed.

Reply:
Changed it to contextId